### PR TITLE
Issue #9: Reduce the number of jobs for 3.5-3.11

### PR DESCRIPTION
**Pull request by jaypha**
Add Moodle plugin CI group for Moodle 3.5-3.11 lite.
Includes full number of jobs for 3.5, 3.9 and 3.11.
Retains only one job for all other versions.
Reduces the number of jobs from 32 to 14.

### CODE REVIEW

**Comment by brendanheywood** [comment]
no, lets change the actual existing workflow, we don't want more

**Comment by keevan** [comment]
Also if you could add a comment to each block of versions you run, with a loose description of why it's running specific versions based on issue #9

File: .github/workflows/group-35-to-311-ci.yml [hunk 1]
```diff
@@ -51,27 +51,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        # Create a mtrix to fully test all combinations for version 3.5 (LTS)
```
Comment (brendanheywood) [inline .github/workflows/group-35-to-311-ci.yml:54]: typo in matrix

File: .github/workflows/group-35-to-311-ci.yml [hunk 2]
```diff
@@ -51,27 +51,24 @@ jobs:
...
         database: ['mariadb', 'pgsql']
-        moodle-branch: ['MOODLE_35_STABLE', 'MOODLE_36_STABLE', 'MOODLE_37_STABLE', 'MOODLE_38_STABLE']
+        moodle-branch: ['MOODLE_35_STABLE']
```
Comment (brendanheywood) [inline .github/workflows/group-35-to-311-ci.yml:56]: Would it make more sense to make sure each lts version is in here, and then just the odds in the include?

Whatever we do here is going to be used as the pattern for the others

Comment (brendanheywood) [reply .github/workflows/group-35-to-311-ci.yml:56]: Possibly not as the matrix might include different versions of node or php across older versions

Comment (jaypha) [reply .github/workflows/group-35-to-311-ci.yml:56]: 3.5 and 3.9 have different PHP requirements, so they are not going to fit in the same matrix.

**Pull request merged**